src/lib/components/ui/ScrollView/ScrollView.jsx

```diff
@@ -0,0 +1,130 @@
+
+  const scrollPositionStart = horizontal ? 'left' : 'top';
+  const scrollPositionEnd = horizontal ? 'right' : 'bottom';
+  const scrollViewContentEl = useRef(null);
+  const scrollViewViewportEl = useRef(null);
+
+  const setAlpha = (rgba, alpha) => {
+    const newColor = { ...rgba };
+    newColor.alpha = alpha;
+    return newColor;
+  };
+
+  const rgbaToString = (rgba) => (
+    `rgba(${rgba.red}, ${rgba.green}, ${rgba.blue}, ${rgba.alpha})`
+  );
+
+  const handleScrollViewState = (currentPosition) => {
+    const isScrolledAtStartActive = currentPosition[scrollPositionStart] < 0;
+    const isScrolledAtEndActive = currentPosition[scrollPositionEnd] > 0;
+
+    if (isScrolledAtStartActive !== isScrolledAtStart) {
+      setIsScrolledAtStart(isScrolledAtStartActive);
+    }
+
+    if (isScrolledAtEndActive !== isScrolledAtEnd) {
+      setIsScrolledAtEnd(isScrolledAtEndActive);
+    }
+  };
+
+  const shadowStartColor = rgbaToString(shadowColor);
+  const shadowEndColor = rgbaToString(setAlpha(shadowColor, 0));
+
+  useLoadResize(
+    (currentPosition) => (handleScrollViewState(currentPosition)),
+    [isScrolledAtStart, isScrolledAtEnd],
+    scrollViewContentEl,
+    scrollViewViewportEl,
+    100,
+  );
+
+  useScrollPosition(
+    (currentPosition) => (handleScrollViewState(currentPosition)),
+    [isScrolledAtStart, isScrolledAtEnd],
+    scrollViewContentEl,
+    scrollViewViewportEl,
+    50,
+  );
+
+  return (
+    <div
+      className={[
+        styles.root,
+        isScrolledAtStart ? styles.isScrolledAtStart : '',
+        isScrolledAtEnd ? styles.isScrolledAtEnd : '',
+        horizontal ? styles.isRootHorizontal : styles.isRootVertical,
+      ].join(' ')}
+    >
+      <div className={styles.viewport} ref={scrollViewViewportEl}>
+        <div className={styles.content} ref={scrollViewContentEl}>
+          {children}
+        </div>
+      </div>
+      <div
+        className={styles.shadowStart}
+        style={{
+          backgroundImage: `linear-gradient(${horizontal ? 'to right' : 'to bottom'}, ${shadowStartColor}, ${shadowEndColor})`,
+          height: horizontal ? 'auto' : shadowSize,
+          width: horizontal ? shadowSize : 'auto',
+        }}
+        aria-hidden
+      />
+      <div
+        className={styles.shadowEnd}
+        style={{
+          backgroundImage: `linear-gradient(${horizontal ? 'to left' : 'to top'}, ${shadowStartColor}, ${shadowEndColor})`,
+          height: horizontal ? 'auto' : shadowSize,
+          width: horizontal ? shadowSize : 'auto',
+        }}
+        aria-hidden
+      />
+    </div>
+  );
+};
+
+ScrollView.defaultProps = {
+  horizontal: false,
+  shadowColor: {
+    alpha: 1,
+    blue: 255,
+    green: 255,
+    red: 255,
+  },
+  shadowSize: '2em',
+};
+
+ScrollView.propTypes = {
+  children: PropTypes.oneOfType([
+    PropTypes.arrayOf(PropTypes.node),
+    PropTypes.node,
+  ]).isRequired,
+  horizontal: PropTypes.bool,
```

Comment:
I think, I would prefer to have here it expressed as prop `direction` that would take values `horizontal` and `vertical`. 

This makes it seem like horizontal is special case of vertical.

Reply:
OK, changed to `direction` prop.